### PR TITLE
ODATA-1442 - Update Aggregation ABNF with changes from previous issues

### PR DESCRIPTION
**Pull request by GeraldKrause**

### CODE REVIEW

**Review by GeraldKrause — COMMENTED** [review]
Looks good to me

**Review by HeikoTheissen — CHANGES_REQUESTED** [review]
Remove test case
https://github.com/oasis-tcs/odata-abnf/blob/eea041541140d958da16e67edae1f7bdbd44cb53/abnf/odata-aggregation-testcases.xml#L274-L276
as a consequence of ODATA-1427.

File: abnf/odata-aggregation-abnf.txt [hunk 1]
```diff
@@ -79,28 +82,37 @@ primitiveProperty =/ expressionAlias / customAggregate
 apply      = ( "$apply" / "apply" ) EQ applyExpr
 applyExpr  = applyTrafo *( "/" applyTrafo )
 applyTrafo = aggregateTrafo
+           / topcountTrafo
```
Comment (sthagen) [inline abnf/odata-aggregation-abnf.txt:85]: Why give up on the lexical ordering - maybe add the semantics driving the new ordering as comment?

File: obsolete/odata-aggregation-testcases.xml [hunk 2]
```diff
@@ -1,491 +1,649 @@
...
-    <Input>$apply=groupby((Continent/Name,Country/Name),aggregate(Population with sum as CountryPopulation))/concat(filter(CountryPopulation ge 10000000),groupby((Continent/Name),aggregate(CountryPopulation with sum as TotalPopulation)))</Input>
-  </TestCase>
-  <TestCase Name="transformation sequences - filter and expand" Rule="queryOptions">
-    <Input>$apply=filter(Status eq 'incomplete')/expand(Items,filter(not Shipped))/groupby((Customer/Country),aggregate(Items(Amount with sum as ItemAmount)))</Input>
-  </TestCase>
-  <TestCase Name="aggregate in $expand" Rule="odataRelativeUri">
-    <Input>Categories?$expand=Products($apply=aggregate(Price with average as AveragePrice))</Input>
-  </TestCase>
-</TestSuite>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sthagen) [inline obsolete/odata-aggregation-testcases.xml:1]: Are these space changes (like line ending)? ... I cannot see that with how I operate the GitHub browser diff UI, sorry.

**Review by sthagen — CHANGES_REQUESTED** [review]
Please kindly consider amending w.r.t. the ABNF "broken" ordering of alternatives comment; otherwise LGTM.

**Review by HeikoTheissen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]
LGTM

File: abnf/odata-aggregation-abnf.txt (hunk 1 above)
Comment (ralfhandl) [reply abnf/odata-aggregation-abnf.txt]: Fixed: now ordered lexicographically

**Review by sthagen — APPROVED** [review]
Nice. Thanks.

**Pull request merged**